index.js

```diff
@@ -4,23 +4,45 @@
  * @param {Object} store - An object representing a single store. See the instructions for details on its shape.
  * @returns {Object[]} The same `stores` array that was inputted.
  */
-function addNewStore(stores, store) {}
+function addNewStore(stores, store) {
+    stores.push(store);
+  
+  return stores
+}
 
 /**
  * Removes a store object at the given position.
  * @param {Object[]]} stores - An array of store objects.
  * @param {number} index - A number representing the index of the store to be removed from the array.
  * @returns {Object[]} The same `stores` array that was inputted.
  */
-function removeStoreAtPosition(stores, index) {}
+function removeStoreAtPosition(stores, index) {
+      stores.splice(index,1)  
+  return stores
+
+}
 
 /**
  * Creates a duplicate of the `store` object. No references should be shared between the inputted `store` and the result.
  * @param {Object} store - An object representing a single store. See the instructions for details on its shape.
  * @returns {Object} The duplicated store object. This should not be the same as the store that was inputted.
  */
-function duplicateStore(store) {}
+function duplicateStore(store) {
+  let copy = {
+    name: store.name,
+  boardGames: [],
+  address: {
+  }}
+  for (let game of store.boardGames){
+  copy.boardGames.push(game)
+  }
+  for (let property in store.address){
```

Comment:
how can you use the for in loop to dynamically create the keys and values in your copy object?